subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperGenerationIntegrationTest.groovy

```diff
@@ -59,6 +60,22 @@ class WrapperGenerationIntegrationTest extends AbstractIntegrationSpec {
         file("gradle/wrapper/gradle-wrapper.properties").text.contains("distributionUrl=https\\://services.gradle.org/distributions/gradle-2.2.1-bin.zip")
     }
 
+    def "generated wrapper scripts for given distribution type from command-line"() {
+        when:
+        run "wrapper", "--gradle-version", "2.13", "--dist-type", "all"
+
+        then:
+        file("gradle/wrapper/gradle-wrapper.properties").text.contains("distributionUrl=https\\://services.gradle.org/distributions/gradle-2.13-all.zip")
+    }
+
+    def "generated wrapper scripts for given distribution type from command-line"() {
+        when:
+        run "wrapper", "--gradle-version", "2.13", "--dist-type", "invalid-distribution-type"
+
+        then:
+        thrown(CommandLineArgumentException)
```

Comment:
Found it out, needed to use `failure.assertHasDescription()` instead of `failure.assertHasCause()`.